x-pack/plugins/enterprise_search/public/applications/app_search/components/credentials/credentials_logic.ts

```diff
@@ -0,0 +1,294 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { kea, MakeLogicType } from 'kea';
+
+import { formatApiName } from '../../utils';
+import { ADMIN, PRIVATE } from '../../constants/credentials';
+
+import { HttpLogic } from '../../../shared/http';
+import { IApiToken, IEngine, ICredentialsDetails } from '../../../../../common/types/app_search';
+import { IMeta } from '../../../../../common/types';
+
+const defaultApiToken: IApiToken = {
+  name: '',
+  type: PRIVATE,
+  read: true,
+  write: true,
+  access_all_engines: true,
+};
+
+// TODO CREATE_MESSAGE, UPDATE_MESSAGE, and DELETE_MESSAGE from ent-search
+
+export const apiTokenSort = (apiTokenA: IApiToken, apiTokenB: IApiToken) => {
+  if (!apiTokenA.id) {
+    return -1;
+  }
+  if (!apiTokenB.id) {
+    return 1;
+  }
+  return apiTokenA.id - apiTokenB.id;
+};
+
+export interface ICredentialsLogicActions {
+  addEngineName(engineName: string): string;
+  onApiKeyDelete(tokenName: string): string;
+  onApiTokenCreateSuccess(apiToken: IApiToken): IApiToken;
+  onApiTokenError(formErrors: string[]): string[];
+  onApiTokenUpdateSuccess(apiToken: IApiToken): IApiToken;
+  removeEngineName(engineName: string): string;
+  setAccessAllEngines(accessAll: boolean): boolean;
+  setCredentialsData(meta: IMeta, apiTokens: IApiToken[]): { meta: IMeta; apiTokens: IApiToken[] };
+  setCredentialsDetails(details: ICredentialsDetails): ICredentialsDetails;
+  setNameInputBlurred(isBlurred: boolean): boolean;
+  setTokenReadWrite({
+    name,
+    checked,
+  }: {
+    name: string;
+    checked: boolean;
+  }): {
+    name: string;
+    checked: boolean;
+  };
+
+  setTokenName(name: string): string;
+  setTokenType(tokenType: string): string;
+  toggleCredentialsForm(apiToken?: IApiToken): IApiToken;
+  hideCredentialsForm(): { value: boolean };
+  resetCredentials(): { value: boolean };
+  initializeCredentialsData(): { value: boolean };
+  fetchCredentials(page?: number): number;
+  fetchDetails(): { value: boolean };
+  deleteApiKey(tokenName: string): string;
+  onApiTokenChange(): { value: boolean };
+  onEngineSelect(engineName: string): string;
+}
+
+export interface ICredentialsLogicValues {
+  activeApiToken: IApiToken;
+  activeApiTokenIsExisting: boolean;
+  activeApiTokenRawName: string;
+  apiTokens: IApiToken[];
+  apiUrl: string;
```

Comment:
Tech debt, not a blocker/change request unless you really like the idea of it - we can potentially lose this value since it's a repeat of `enterpriseSearchUrl` (already stored in our KibanaContext). No need to fetch it/store it separately in CredentialsLogic

Reply:
Nice, I didn't realize that.

Reply:
Removed apiUrl d1c2444c82b8cd0ac74afbdf583bc1c3a3654379